src/Renderer/ThreeExtended/OBB.js

```diff
@@ -175,41 +173,55 @@ OBB.extentToOBB = function extentToOBB(extent, minHeight = 0, maxHeight = 0) {
     tmp.cardinals[7]._values[0] = phiStart;
     tmp.cardinals[7]._values[1] = thetaStart + bboxDimension.y * 0.5;
 
-    var cardin3DPlane = [];
+    const cardinalsXYZ = [];
+    const centersLongitude = tmp.cardinals[8].longitude(UNIT.RADIAN);
+    for (var i = 0; i < tmp.cardinals.length; i++) {
+        cardinalsXYZ.push(tmp.cardinals[i].as('EPSG:4978').xyz());
+    }
+
+    return this.cardinalsXYZToOBB(cardinalsXYZ, centersLongitude, true, minHeight, maxHeight);
+};
 
+OBB.cardinalsXYZToOBB = function cardinalsXYZToOBB(cardinals, centerPhi, isEllipsoid, minHeight = 0, maxHeight = 0) {
     tmp.maxV.set(-1000, -1000, -1000);
     tmp.minV.set(1000, 1000, 1000);
-    var halfMaxHeight = 0;
+
+    let halfMaxHeight = 0;
+    tmp.normal.copy(cardinals[8]).normalize();
     tmp.tangentPlaneAtOrigin.set(tmp.normal, 0);
 
     tmp.planeZ.setFromUnitVectors(tmp.normal, tmp.zUp);
     tmp.qRotY.setFromAxisAngle(
-        new THREE.Vector3(0, 0, 1), -tmp.cardinals[8].longitude(UNIT.RADIAN));
+        new THREE.Vector3(0, 0, 1), -centerPhi);
     tmp.qRotY.multiply(tmp.planeZ);
 
-    for (var i = 0; i < tmp.cardinals.length; i++) {
-        tmp.cardinals[i].as('EPSG:4978', tmp.epsg4978).xyz(tmp.cardinal3D);
-        cardin3DPlane.push(tmp.tangentPlaneAtOrigin.projectPoint(tmp.cardinal3D));
-        const d = cardin3DPlane[i].distanceTo(tmp.cardinal3D.sub(centerWorld));
+    let point5InPlaneX;
+    for (let i = 0; i < cardinals.length; i++) {
```

Comment:
you could use `for (const ... of ...)` ?

Reply:
I use the `i` in the for loop so I can't use `for of`